### ISSUE TITLE
In android Q touch and swipe is not working with Minitouch(libedev)

### ISSUE THREAD

**Issue by varundtsfi** [openstf/minicap#186]
i try on android Q，

log： 
**Unable to open** device /dev/input/event7 for inspectionopen: 

**Permission denied Unable to open device** /dev/input/event6 for inspectionopen: Permission denied 

Unable to open device /dev/input/event5 for inspectionopen: Permission denied Unable to open device /dev/input/event1 for inspectionopen: Permission denied 

Unable to open device /dev/input/event3 for inspectionopen: Permission denied Unable to open device /dev/input/mice for inspectionopen: Permission denied 

Unable to open device /dev/input/event0 for inspectionopen: Permission denied Unable to open device /dev/input/event2 for inspectionopen: Permission denied 

Unable to open device /dev/input/event4 for inspectionUnable to find a suitable touch device

I am using Pixel Device Android Q Beta 6.

Android Q has no **"USB debugging (Security settings)** like Xiaomi Device having. 

Any Suggestion?

**Comment by XiaLeon** [comment]
@varundtsfi  Are you sure there's no "USB Debugging" option in the "Developer Options" menu ? If there isn't, just click the "Software Version"  7 times and it'll show up.

**Comment by varundtsfi** [comment]
> @varundtsfi Are you sure there's no "USB Debugging" option in the "Developer Options" menu ? If there isn't, just click the "Software Version" 7 times and it'll show up.
> It said the Permission is denied, so I suppose you didn't root your device

It's there but there is another option called   "USB debugging (Security settings) like Xiaomi Device having to simulate input events..

Any Suggestion?

**Comment by XiaLeon** [comment]
@varundtsfi upload a screen shot and I'll take a look

**Comment by varundtsfi** [comment]
> @varundtsfi upload a screen shot and I'll take a look

Hi @XiaLeon , 

I have attached the logs for your information but I have debugged the issue and found android 10 beta input drivers are not allowing to read the input files.
**SELinux is blocking minitouch(wrapper on libedev) from accessing them.**

e.g:
**Unable to open device /dev/input/event1 for inspectionopen: Permission denied 
Unable to open device /dev/input/event2 for inspectionopen: Permission denied**

Google has disabled access to the input devices in the past and then allowed it again in later betas. We'll see I think in android 10 beta third party libs are not supported.

But for now I have found a solution. We can use android standard library to handle the touch events. And there are very less chances that android will ever depreciate these services.

[](https://developer.android.com/reference/android/view/MotionEvent.html)

By the above implementation I am able to handle the touch events and its working fine.
